clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py

```diff
@@ -0,0 +1,650 @@
+
+
+def find_clinical_data(
+    clinical_data_directory: PathLike,
+) -> List[DataFrame]:
+    """Finds the clinical data associated with the dataset.
+
+    Parameters
+    ----------
+    clinical_data_directory: PathLike
+        Path to the clinical data.
+
+    Returns
+    -------
+    List[DataFrame]
+        Dataframes containing the clinical data
+    """
+    return (
+        _read_file(_check_file(clinical_data_directory, pattern))
+        for pattern in (
+            "FINAL*DEMOGRAPHICS*.xlsx",
+            "FINAL*IMAGING*.xlsx",
+            "FINAL*CLINICAL*.xlsx",
+        )
+    )
+
+
+def complete_clinical_data(
+    df_demographics: DataFrame, df_imaging: DataFrame, df_clinical: DataFrame
+) -> DataFrame:
+    """Merges the different clincal dataframes into one.
+
+    Parameters
+    ----------
+    df_demographics: DataFrame
+        Dataframe containing the demographic data
+
+    df_imaging: DataFrame
+        Dataframe containing the imaging data
+
+    df_clinical: DataFrame
+        Dataframe containing the clinical data
+
+    Returns
+    -------
+    df_clinical_complete: DataFrame
+        Dataframe with the data of the 3 input dataframes
+    """
+    df_clinical_complete = df_imaging.merge(
+        df_demographics, how="inner", on=["blinded_code", "blinded_site", "visit"]
+    ).drop(columns="diagnosis")
+    df_clinical = df_clinical.dropna(subset=["blinded_code", "blinded_site", "visit"])
+    df_clinical_complete = df_clinical_complete.merge(
+        df_clinical[
+            [
+                "blinded_code",
+                "blinded_site",
+                "visit",
+                "diagnosis",
+                "ftld-cdr-global",
+                "cdr-sob",
+            ]
+        ],
+        how="inner",
+        on=["blinded_code", "blinded_site", "visit"],
+    )
+    return df_clinical_complete
+
+
+def dataset_to_bids(complete_data_df: DataFrame, gif: bool) -> Dict[str, DataFrame]:
+    """Selects the data needed to write the participants, sessions, and scans tsvs.
+
+    Parameters
+    ----------
+    complete_data_df: DataFrame
+        Dataframe containing the merged data extracted from the raw images and the clinical data
+
+    gif: bool
+        If True, indicates the user wants to have the values of the gif parcellation
+
+    Returns
+    -------
+    Dict[str, DataFrame]
+        Dictionary containing as key participants, sessions and scans, and the values wanted for each tsv
+    """
+    import os
+
+    # generates participants, sessions and scans tsv
+    complete_data_df = complete_data_df.set_index(
+        ["participant_id", "session_id", "modality", "bids_filename"],
+        verify_integrity=True,
+    )
+    # open the reference for building the tsvs:
+    path_to_ref_csv = os.path.join(
+        os.path.dirname(os.path.dirname(os.path.dirname(__file__))),
+        "data",
+        "genfi_ref.csv",
+    )
+    df_ref = pd.read_csv(path_to_ref_csv, sep=";")
+    if not gif:
```

Comment:
I might be missing something, but this is the only spot where gif is used in this function.
Is the print worth the additional parameter ?
Also, it doesn't seem to do what the doc says it should, right ?

Reply:
yes the purpose was not the print aha. I forgot to finalize this function... On it right now